components/script/dom/xmlhttprequest.rs

```diff
@@ -1527,6 +1527,7 @@ impl XMLHttpRequest {
             docloader,
             None,
             None,
+            doc.status_code(),
```

Comment:
Do we actually want to propagate the status code from the original document like this? That seems like it could be confusing, since it's a synthetic document. Should it be the status code of the xhr response instead?